tensorbay/opendataset/VOC2012Action/loader.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name, missing-module-docstring
+
+import os
+import typing
+
+import xmltodict
+
+from tensorbay.dataset import Data, Dataset
+from tensorbay.label import LabeledBox2D
+
+_SEGMENT_NAMES = (
+    "train",
+    "trainval",
+    "val",
+)
+DATASET_NAME = "VOC2012Action"
+
+
+def VOC2012Action(path: str) -> Dataset:
+    """Dataloader of the 'VOC2012Action'_ dataset.
+
+    .. _VOC2012Action: http://host.robots.ox.ac.uk/pascal/VOC/voc2012/
+
+    The file structure should be like::
+
+        <path>
+            Annotations/
+                <image_name>.xml
+                ...
+            JPEGImages/
+                <image_name>.jpg
+                ...
+            ImageSets/
+                Action/
+                    train.txt
+                    trainval.txt
+                    val.txt
+                    ...
+                ...
+            ...
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class: `~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    paths = {
+        "Annotations": os.path.join(path, "Annotations"),
+        "Image": os.path.join(path, "JPEGImages"),
+        "Action": os.path.join(path, "ImageSets", "Action"),
+    }
+
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog("catalog.json")
+
+    for segment_name in _SEGMENT_NAMES:
+        segment = dataset.create_segment(segment_name)
+        with open(os.path.join(paths["Action"], f"{segment_name}.txt")) as fp:
+            for filename in fp:
+                filename = filename.strip()
+                segment.append(_get_data(filename, paths))
+    return dataset
+
+
+def _get_data(filename: str, paths: typing.Dict[str, str]) -> Data:
+    """Get all information of the datum corresponding to filename.
+
+    Arguments:
+        filename: The filename of the data.
+        paths: The dictionary includes paths.
+
+    Returns:
+        Data: class: `~tensorbay.dataset.data.Data` instance.
+
+    """
+    data = Data(os.path.join(paths["Image"], f"{filename}.jpg"))
+    data.label.box2d = []
+    with open(os.path.join(paths["Annotations"], f"{filename}.xml"), "r") as fp:
+        objects = xmltodict.parse(fp.read())["annotation"]["object"]
+        if not isinstance(objects, list):
```

Comment:
why does the type of `objects` needs to be checked?

Reply:
it may be a OrderedDict